lzy/worker/src/main/java/ai/lzy/worker/env/DockerEnvironment.java

```diff
@@ -262,19 +260,14 @@ public BaseEnvConfig config() {
     }
 
     private void prepareImage(String image, StreamQueue.LogHandle handle) throws Exception {
-        var msg = new AtomicReference<>("Pulling image " + image + " ...");
-        handle.logOut(msg.get());
+        handle.logOut("Pulling image {} ...", image);
         AtomicInteger pullingAttempt = new AtomicInteger(0);
         retry.executeCallable(() -> {
             LOG.info("Pulling image {}, attempt {}", image, pullingAttempt.incrementAndGet());
             final var pullingImage = client
                 .pullImageCmd(config.image())
                 .exec(new PullImageResultCallback());
-            while (!pullingImage.awaitCompletion(1, TimeUnit.SECONDS)) {
```

Comment:
It doesn't work. awaitCompletion method will end pulling after timeout. 
https://github.com/docker-java/docker-java/blob/dd0de7c83c8a2691d96c50edbf5c48f1bced0c7c/docker-java-api/src/main/java/com/github/dockerjava/api/async/ResultCallbackTemplate.java#L116